diesel/src/connection/transaction_manager.rs

```diff
@@ -56,10 +56,16 @@ pub trait TransactionManager<Conn: Connection> {
                 Self::commit_transaction(conn)?;
                 Ok(value)
             }
-            Err(e) => {
-                Self::rollback_transaction(conn).map_err(|e| Error::RollbackError(Box::new(e)))?;
-                Err(e)
-            }
+            Err(user_error) => Err(match Self::rollback_transaction(conn) {
+                Ok(()) => user_error,
+                Err(rollback_error) => E::from(match rollback_error {
```

Comment:
Can we unnest the match and Err wrapping here? That would make it much easier to read.

Reply:
I typically find this kind of writing easier to read because it quickly reads "whatever happens, we're gonna return an error - then here here's what error it will be".
That being said I can change that to wrap each error in err individually if you have a strong preference for this.